client/grpc/grpc.go

```diff
@@ -432,19 +450,19 @@ func (g *grpcClient) Call(ctx context.Context, req client.Request, rsp interface
 			time.Sleep(t)
 		}
 
-		// select next node
-		node, err := next()
-		service := req.Service()
+		// lookup the route to send the reques to
+		route, err := g.lookupRoute(req, callOpts)
 		if err != nil {
-			if err == selector.ErrNotFound {
-				return errors.InternalServerError("go.micro.client", "service %s: %s", service, err.Error())
-			}
-			return errors.InternalServerError("go.micro.client", "error selecting %s node: %s", service, err.Error())
+			return err
 		}
 
 		// make the call
-		err = gcall(ctx, node, req, rsp, callOpts)
-		g.opts.Selector.Mark(service, node, err)
+		err = gcall(ctx, route, req, rsp, callOpts)
+
+		// record the result of the call to inform future routing decisions
+		g.opts.Selector.Record(route, err)
```

Comment:
Why do we bother recording this? I thought the selector isn't involved with determining the route anyway.

Reply:
The selector is responsible for picking one of the routes. Right now we're using the default (random) selector, but in the future we'll implement a more intelligent selector which uses recent errors / response times to determine which route is optimal.